gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -838,6 +838,26 @@ void GazeboRosCamera::OnTrigger(const std_msgs::msg::Empty::SharedPtr)
   impl_->triggered++;
 }
 
+const std::vector<gazebo::rendering::CameraPtr> GazeboRosCamera::GetCameras() const
+{
+  return impl_->camera_;
+}
+
+const gazebo_ros::Node::SharedPtr GazeboRosCamera::GetRosNode() const
+{
+  return impl_->ros_node_;
```

Comment:
just curious, why do you need the ROS node pointer? are you planning to create extra publishers?

Reply:
Yes my subclass creates a subscriber to toggle the lens caps. Do you have a better solution in mind?

Reply:
I'm not sure if it works, but you might be able to call `gazebo_ros::Node::Get(_sdf);` in the subclasses `Load` method. Then you'll also be able to parse any custom SDF tags specific to your plugin.

Reply:
@scpeters @jacobperron Thanks for the suggestion. I can confirm that `gazebo_ros::Node::Get(_sdf);` does work. I have updated the PR to remove the ros node accessor.